services/web/server/src/simcore_service_webserver/statics.py

```diff
@@ -75,66 +147,33 @@ async def get_frontend_ria(request: web.Request):
-    )
 
+    cached_indexes: Dict[str, str] = request.app[APP_FRONTEND_CACHED_INDEXES_KEY]
+    if target_frontend not in cached_indexes:
+        raise web.HTTPNotFound()
 
-def create_statics_settings(app) -> Dict:
-    # Adds general server settings
-    info: Dict = app[APP_SETTINGS_KEY].to_client_statics()
+    body = cached_indexes[target_frontend]
+    return web.Response(body=body, content_type="text/html")
 
-    # Adds specifics to front-end app
-    info.update(FrontEndAppSettings().to_statics())
 
-    return info
-
-
-async def _start_statics(app: web.Application):
-    # NOTE: in devel model, the folder might be under construction
-    # (qx-compile takes time), therefore we create statics.json
-    # on_startup instead of upon setup
-
-    resource_dir: Path = app[APP_STATICS_OUTDIR_KEY] / "resource"
-    statics_settings: Dict = create_statics_settings(app)
-
-    @retry(
-        wait=wait_random(min=1, max=3),
-        stop=stop_after_attempt(3),
-        after=after_log(log, logging.WARNING),
-    )
-    async def do_write_statics_file() -> None:
-        with open(resource_dir / "statics.json", "wt") as fh:
-            json.dump(statics_settings, fh)
-
-    # Creating static info
-    await do_write_statics_file()
+async def get_statics_json(request: web.Request):  # pylint: disable=unused-argument
+    statics_json = request.app[APP_FRONTEND_CACHED_STATICS_JSON_KEY]
+    return web.Response(body=statics_json, content_type="application/json")
 
 
 @app_module_setup(__name__, ModuleCategory.SYSTEM, logger=log)
-def setup_statics(app: web.Application):
-
-    # NOTE: source-output and build-output have both the same subfolder structure
-    cfg = app[APP_CONFIG_KEY]["main"]
-    app[APP_STATICS_OUTDIR_KEY] = statics_dir = Path(cfg["client_outdir"]).expanduser()
-
-    # Creating static routes
-    routes = web.RouteTableDef()
-    is_dev: bool = app[APP_SETTINGS_KEY].build_target in [None, "development"]
-    for name in STATIC_DIRNAMES:
-        folder = statics_dir / name
-
-        # avoids problems restarting when qx-compile takes longer to product outputs
-        if not folder.exists() and is_dev:
-            os.makedirs(folder, exist_ok=True)
-
-        # can navigate file index in dev mode
-        routes.static(f"/{folder.name}", folder, show_index=is_dev)
-    app.add_routes(routes)
-
-    # Create dynamic route to serve front-end client
-    app.router.add_get("/", get_frontend_ria, name=INDEX_RESOURCE_NAME)
-
-    # Delayed creation of statics.json (mostly for dev mode)
-    app.on_startup.append(_start_statics)
+def setup_statics(app: web.Application) -> None:
+    settings = assemble_settings(app)
+    if not settings.enabled:
+        log.warning("Static webserver module is disbaled")
+        return
+
+    # register routes to solve the correct frontend
+    app.router.add_get("/", get_cached_frontend_index, name=INDEX_RESOURCE_NAME)
```

Comment:
how does this work now?
the client accesses the root / and the webserver serves the index.html? then it goes into the static server?

Reply:
All static content routes: `/resource`, `/transpiled`, `/s4l`, `/osparc`, `/tis` are served by the `static-webserver`
https://github.com/ITISFoundation/osparc-simcore/blob/d9f85d9c98e33a039c6fe5d2cb44038dad7df62b/services/docker-compose.yml#L138

The `webserver` will only serve: `/v0`, `/`, `/socket.io` and `/static-frontend-data.json`
https://github.com/ITISFoundation/osparc-simcore/blob/d9f85d9c98e33a039c6fe5d2cb44038dad7df62b/services/docker-compose.yml#L179

When the `webserver` starts it will prepare and cache the static 2 routes called static:
- `/` content is composed by making 3 http requests (once for each product) to the `index.html` in each of the 3 product directories `/osparc`, `/tis`, `/s4l` replacing rules are applied and the content is stored. Each `index.html` is a very small file containing links to the rest of the content which will be served by the `static-webserver`
- `/static-frontend-data.json` contains information which is used by the frontend to properly render the client

Reply:
:+1:

Reply:
👍 please add some of the doc above next to the code that assigns routes